sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_condition.py

```diff
@@ -37,12 +38,15 @@ def pipeline_func(int_param1: int, int_param2: int):
 
     :param condition: The condition of the execution flow.
         The value could be a boolean type control output or a pipeline expression.
-    :type condition: Union[str, bool, InputOutputBase]
+    :type condition: Union[str, bool, InputOutputBase, PipelineExpression]
     :param true_block: Block to be executed if condition resolved result is true.
     :type true_block: BaseNode
     :param false_block: Block to be executed if condition resolved result is false.
     :type false_block:  BaseNode
     :return: The condition node.
     :rtype: ConditionNode
     """
+    # resolve expression as command component
+    if isinstance(condition, PipelineExpression):
```

Comment:
Seems this looks like the case: if user pass a node with only one output, we can do some similar logic to extract the single output

Reply:
yes, is there any existing method to do so?

Reply:
Can you add this logic here?

Reply:
Refine this part as below:

1. convert `PipelineExpression` to a node
2. check the number of node's output, if not 1, raise `UserErrorException`
3. get the only output of the node

also add type `BaseNode` to param `condition`'s annotation